packages/neon/neon/test/timer_bloc_test.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:neon/neon.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('TimerBloc', () {
+    test('Register timer', () async {
+      const duration = Duration(milliseconds: 100);
+
+      final stopwatch = Stopwatch()..start();
+      TimerBloc().registerTimer(duration, stopwatch.stop);
+      await Future.delayed(duration);
+
+      expect(stopwatch.elapsedMilliseconds, greaterThan(duration.inMilliseconds));
+      expect(stopwatch.elapsedMilliseconds, lessThan(duration.inMilliseconds * 1.1));
+    });
+
+    test('Unregister timer', () async {
+      const duration = Duration(milliseconds: 100);
+
+      TimerBloc().registerTimer(duration, neverCalled).cancel();
+      await Future.delayed(duration);
```

Comment:
Other than implicitly testing that it doesn't throw something we aren't really testing anything here:
I'd suggest:

```dart
    test('Unregister timer', () async {
      const duration = Duration(milliseconds: 100);
      final callback = neverCalled;

      TimerBloc().registerTimer(duration, callback).cancel();
      await Future.delayed(duration);

      expect(TimerBloc().callbacks[duration.inSeconds], isNot(contains(callback)));
    });
```